lib/reservation.rb

```diff
@@ -0,0 +1,38 @@
+require "date"
+
+module Hotel
+  class Reservation
+    attr_reader :ckin_date, :ckout_date, :date_range_string_array, :num_nights_of_stay, :base_cost
+
```

Comment:
Consider moving the checking for overlapping a date with the reservation to this class, so it could handle more actions that deal with a specific reservation.  It might simplify your manager class. 

Also calculating the cost of a reservation is probably best done as a method just in case the dates change (extremely minor point).